compiler/modules/front/expr.bal

```diff
@@ -1442,11 +1453,37 @@ function genLocalFunctionRef(ExprContext cx, string funcName, Position pos) retu
         if atom != () {
             return [functionRefFromAtom(cx, atom, registerName(ref.reg)), ref.reg];
         }
-        return cx.unimplementedErr("return type projection not implemented", pos);
+        if t:isSubtype(cx.mod.tc, semType, t:FUNCTION) {
+            return [(), ref.reg];
+        }
     }
     return cx.semanticErr("only a value of function type can be called", pos);
 }
 
+
+function finishCodeGenFunctionInexactCall(ExprContext cx, bir:BasicBlock bb, s:FunctionCallExpr expr,
+                                          bir:Register func) returns CodeGenError|ExprEffect {
+    t:Context tc = cx.mod.tc;
+    t:SemType funcTy = func.semType;
+    bir:Operand[] args = [];
+    t:SemType[] argTypes = [];
+    bir:BasicBlock curBlock = bb;
+    foreach s:Expr argExpr in expr.args {
+        var { result: arg, block: nextBlock } = check codeGenExpr(cx, curBlock, (), argExpr);
```

Comment:
Need to project on the domain to get the contextually expected type.

Reply:
Fixed